homeassistant/components/doorbird.py

```diff
@@ -173,8 +275,49 @@ class DoorbirdRequestView(HomeAssistantView):
     @asyncio.coroutine
     def get(self, request, sensor):
         """Respond to requests from the device."""
+        from aiohttp import web
         hass = request.app['hass']
 
         hass.bus.async_fire('{}_{}'.format(DOMAIN, sensor))
 
-        return 'OK'
+        return web.Response(status=200, text='OK')
+
+
+class DoorBirdCleanupView(HomeAssistantView):
+    """Provide a URL to call to delete ALL webhooks/schedules."""
+
+    requires_auth = False
+    url = API_URL + '/clear/{name}'
+    name = 'DoorBird Cleanup'
+
+    # pylint: disable=no-self-use
+    async def get(self, request, name):
+        """Act on requests."""
+        from aiohttp import web
+        hass = request.app['hass']
+
+        # Find matching device
+        for config_device in hass.data[DOMAIN]:
+            if config_device.name == name:
+                return self._clear(config_device.device)
+
+        # No matching device
+        return web.Response(status=404,
+                            text='Device "{}" not found'.format(name))
+
+    @staticmethod
+    def _clear(device):
+        from aiohttp import web
+
+        # Clear webhooks
+        favorites = device.favorites()
+        for favorite_type in favorites:
+            for favorite_id in favorites[favorite_type]:
+                device.delete_favorite(favorite_type, favorite_id)
```

Comment:
Is does doing I/O? You're calling this from an async method which does not allow doing I/O